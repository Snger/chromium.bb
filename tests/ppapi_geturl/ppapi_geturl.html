<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html>
  <!-- Copyright 2010 Google Inc.  All rights reserved. -->
  <head>
    <META HTTP-EQUIV="Pragma" CONTENT="no-cache" />
    <META HTTP-EQUIV="Expires" CONTENT="-1" />
    <style type="text/css">
      pre.notrun { background-color: skyblue }
      pre.pass { background-color: lime }
      pre.fail { background-color: red }
    </style>
    <script type="application/x-javascript">
      //<![CDATA[
var plugin = 0;
var TOTAL_TESTS = 6;
var testsNotRun = TOTAL_TESTS;
var testsPassed = 0;
var testTimeoutHandler = 0;
var testTimedOut = false;

// Each test will load the url using 2 approaches.
var LOAD_VIA_BUFFER = 0;
var LOAD_VIA_FILE = 1;

// Test1
var logValid = new Array();
var VALID_URL = 'ppapi_geturl_success.html';
// Test2
var logCrossOrigin = new Array();
var CROSSORIGIN_URL = 'http://www.google.com/robots.txt';
// Test3
var logInvalid = new Array();
var INVALID_URL = 'ppapi_nonexistent_url.html';

function isMatch(actual, expected) {
  return (0 <= actual.indexOf(expected));
}

function logTestMessage(logElement, message) {
  logElement.textContent += message + '\n';
  console.log(message + '\n');
}

function setTestStatus(logElement, success, expected, isRightResult) {
  logTestMessage(logElement, 'LOAD STATUS:' +
                 ' success=' + success +
                 ' expected=' + expected +
                 ' result=' + isRightResult);
  if (success != expected || !isRightResult) {
    logElement.className = 'fail';
  } else {
    logElement.className = 'pass';
    ++testsPassed;
  }
}

function getLogElement(logElementArray, asFile) {
  if (asFile) {
    return logElementArray[LOAD_VIA_FILE];
  } else {
    return logElementArray[LOAD_VIA_BUFFER];
  }
}

// Validates the data loaded from the url. To be called by the plugin to
// complete each test on both success and failure or the test will time out.
var ReportResult = function(url, asFile, result, success) {
  --testsNotRun;
  var logElement;
  var expectedSuccess;
  var isRightResult;

  if (isMatch(url, VALID_URL)) {
    expectedSuccess = true;
    isRightResult = isMatch(result, 'TEST PASSED');  // Validate file body.
    logElement = getLogElement(logValid, asFile);
  }
  if (isMatch(url, CROSSORIGIN_URL)) {
    expectedSuccess = false;
    isRightResult = isMatch(result, 'PP_ERROR_NOACCESS');
    logElement = getLogElement(logCrossOrigin, asFile);
  }
  if (isMatch(url, INVALID_URL)) {
    expectedSuccess = false;
    isRightResult = isMatch(result, '404');
    logElement = getLogElement(logInvalid, asFile);
  }
  logTestMessage(logElement, 'FULL URL   : ' + url);
  setTestStatus(logElement, success, expectedSuccess, isRightResult);
  logTestMessage(logElement, 'LOAD RESULT: ' + result);

  if (0 == testsNotRun){
    if (0 != testTimeoutHandler) {
      clearTimeout(testTimeoutHandler);
      testTimeoutHandler = 0;
    }
  }
}

// Test watchdog timer. Makes sure the test does not run too long if
// something hangs.
function handleTestTimeout() {
  clearTimeout(testTimeoutHandler);
  testTimeoutHandler = 0;
  testTimedOut = true;
}

// Start tests asynchronously. Plugin must call ReportResult() above
// to complete each test on success or failure or the test will time out.

function startTest(logId, logElementArray, type, url, AsFile) {
  logElement = document.getElementById(logId);
  if (AsFile) {
    logElementArray[LOAD_VIA_FILE] = logElement;
  } else {
    logElementArray[LOAD_VIA_BUFFER] = logElement;
  }
  logTestMessage(logElement,  'Loading ' + type + ' URL ' + url);
  plugin.loadUrl(url, AsFile);
}

function startTests() {
  plugin = document.getElementById('nacl');
  testTimeoutHandler = setTimeout('handleTestTimeout()', 10000);

  startTest('LogValid', logValid, 'valid', VALID_URL, false);
  startTest('LogValidFile', logValid, 'valid', VALID_URL, true);
  startTest('LogCrossOrigin',
            logCrossOrigin, 'cross-origin', CROSSORIGIN_URL, false);
  startTest('LogCrossOriginFile',
            logCrossOrigin, 'cross-origin', CROSSORIGIN_URL, true);
  startTest('LogInvalid', logInvalid, 'invalid', INVALID_URL, false);
  startTest('LogInvalidFile', logInvalid, 'invalid', INVALID_URL, true);
}
      //]]>
    </script>
    <title>PPAPI GetURL Test</title>
  </head>
  <body onload="nacllib.waitForModulesAndRunTests();"
        onunload="nacllib.cleanUp();" >
    <h1>PPAPI GetURL Test</h1>

    <table border=5 cellpadding=5%>
      <tr><td> <b>Load via Buffer</b>
          <td> <b>Load via File</b>
      <tr><td><pre id='LogValid' class='notrun'></pre>
          <td><pre id='LogValidFile' class='notrun'></pre>
      <tr><td><pre id='LogCrossOrigin' class='notrun'></pre>
          <td><pre id='LogCrossOriginFile' class='notrun'></pre>
      <tr><td><pre id='LogInvalid' class='notrun'></pre>
          <td><pre id='LogInvalidFile' class='notrun'></pre>
    </table>

    <div id="status">NO-STATUS</div>

    <embed type="application/x-ppapi-nacl-srpc" id="nacl" name="nacl_module"
           src="ppapi_geturl.nexe" width="0" height="0" />

    <script type="text/javascript" src="nacl_js_lib.js"></script>
    <script type="text/javascript">
      //<![CDATA[
      var nacllib = new NaclLib('nacl_module', 'status', 500);

      // Returns true (for "wait") to the NaclLib test driver until all of the
      // tests are complete.  I.e., either all of the file load callbacks have
      // completed successfully or an error has occurred.
      nacllib.wait = function() {  // First call after the module was loaded.
        if (0 == plugin) {
          startTests();
          return true;
        } else if (testTimedOut) {  // Testing took too long.
          return false;
        } else if (0 < testsNotRun) {  // Still waiting for tests to finish.
          return true;
        } else { // Testing is complete.  Allow the test driver to call test().
          return false;
        }
      }

      // Returns the test status to the NaclLib test driver.  This is called
      // by the NaclLib test driver after the wait() method has returned false.
      // I.e., this is called only after all tests are complete.
      nacllib.test = function() {
        // The actual testing is all finished by the time this method is
        // called, so just return the test results.
        if (testsPassed == TOTAL_TESTS) {
          return '';  // success
        } else if (testTimedOut) {
          return 'Test timed out.';
        } else {
          return 'Test failed.';
        }
      }
      //]]>
    </script>
  </body>
</html>
