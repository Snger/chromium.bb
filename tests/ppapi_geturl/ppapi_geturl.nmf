{
  "nexes": {
    "x86-64": "ppapi_geturl_x86-64.nexe",
    "x86-32": "ppapi_geturl_x86-32.nexe",
  }
}
