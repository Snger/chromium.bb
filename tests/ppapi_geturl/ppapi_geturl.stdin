# First register a bunch of dummy replays with zero count
# so that those signatures will get registered during "install_upcalls"
# Note: because of the zero count we do not have to get the parameters right
#
# Enumerating all known methods ahead of time helps in many ways:
# * we can delay registering non zero count replays till later which help
#   readability
# * we get an error if for some reason the nexe calls one of these functions.
#   Sadly, the nexe usually fails silently when a method is missing from a
#   service
#
# On the downside, this is currently replicated and must be updated by hand
# when signatures change.

replay 0 HasProperty:CCC:iC *
replay 0 HasMethod:CCC:iC *
replay 0 GetProperty:CCC:CC *
replay 0 GetAllPropertyNames:CC:iCC *
replay 0 SetProperty:CCCC:C *
replay 0 RemoveProperty:CCC:C *
replay 0 Call:CCiCC:CC *
replay 0 Construct:CiCC:CC *
replay 0 Deallocate:C: *
replay 0 PPB_GetInterface:s:i *

replay 0 PPB_Audio_Dev_Create:ii:i *
replay 0 PPB_Audio_Dev_IsAudio:i:i *
replay 0 PPB_Audio_Dev_GetCurrentConfig:i:i *
replay 0 PPB_Audio_Dev_StopPlayback:i:i *
replay 0 PPB_Audio_Dev_StartPlayback:i:i *

replay 0 PPB_AudioConfig_Dev_CreateStereo16Bit:lii:l *
replay 0 PPB_AudioConfig_Dev_IsAudioConfig:l:i *
replay 0 PPB_AudioConfig_Dev_RecommendSampleFrameCount:i:i *
replay 0 PPB_AudioConfig_Dev_GetSampleRate:l:i *
replay 0 PPB_AudioConfig_Dev_GetSampleFrameCount:l:i *

replay 0 PPB_Core_AddRefResource:i: *
replay 0 PPB_Core_ReleaseResource:i: *
replay 0 PPB_Core_GetTime::d *
replay 0 ReleaseResourceMultipleTimes:ii: *

replay 0 PPB_FileIO_Dev_Create:i:i *
replay 0 PPB_FileIO_Dev_IsFileIO:i:i *
replay 0 PPB_FileIO_Dev_Open:iiii:i *
replay 0 PPB_FileIO_Dev_Read:ilii:Ci *

replay 0 PPB_Graphics2D_Create:iCi:i *
replay 0 PPB_Graphics2D_IsGraphics2D:i:i *
replay 0 PPB_Graphics2D_Describe:i:Cii *
replay 0 PPB_Graphics2D_PaintImageData:iiCC: *
replay 0 PPB_Graphics2D_Scroll:iCC: *
replay 0 PPB_Graphics2D_ReplaceContents:ii: *
replay 0 PPB_Graphics2D_Flush:ii:i *

replay 0 PPB_ImageData_GetNativeImageDataFormat::i *
replay 0 PPB_ImageData_IsImageDataFormatSupported:i:i *
replay 0 PPB_ImageData_Create:iiCi:i *
replay 0 PPB_ImageData_IsImageData:i:i *
replay 0 PPB_ImageData_Describe:i:Chii *

replay 0 PPB_Instance_GetWindowObject:i:C *
replay 0 PPB_Instance_GetOwnerElementObject:i:C *
replay 0 PPB_Instance_BindGraphics:ii:i *
replay 0 PPB_Instance_IsFullFrame:i:i *
replay 0 PPB_Instance_ExecuteScript:iCC:CC *

replay 0 PPB_URLRequestInfo_Create:i:i *
replay 0 PPB_URLRequestInfo_IsURLRequestInfo:i:i *
replay 0 PPB_URLRequestInfo_IsURLResponseInfo:i:i *
replay 0 PPB_URLRequestInfo_SetProperty:iiC:i *
replay 0 PPB_URLRequestInfo_AppendDataToBody:iC:i *
replay 0 PPB_URLRequestInfo_AppendFileToBody:illld:i *

replay 0 PPB_URLLoader_Open:iii:i *
replay 0 PPB_URLLoader_Create:i:i *
replay 0 PPB_URLLoader_FollowRedirect:ii:i *
replay 0 PPB_URLLoader_GetUploadProgress:i:lli *
replay 0 PPB_URLLoader_GetDownloadProgress:i:lli *
replay 0 PPB_URLLoader_ReadResponseBody:iii:Ci *
replay 0 PPB_URLLoader_GetResponseInfo:i:i *
replay 0 PPB_URLLoader_Close:i: *
replay 0 PPB_URLLoader_IsURLLoader:i:i *


replay 0 PPB_URLResponseInfo_IsURLResponseInfo:i:i *
replay 0 PPB_URLResponseInfo_GetProperty:ii:C *
replay 0 PPB_URLResponseInfo_GetBodyAsFileRef:i:l *

######################################################################
## END OF PROLOG
######################################################################

# define a bunch of variables for readability

set_variable MODULE i(4444)
set_variable INSTANCE i(5555)

# Tags used by pepper/srpc to marshall js parameters
set_variable JS_STR 4:5
set_variable JS_INT 4:2

# NOTE: the strings below must not contain unescaped spaces or commas
set_variable ARG_loadUrl ${JS_STR},4:7,"loadUrl",1:0
set_variable ARG_ReportResult ${JS_STR},4:12,"ReportResult",4:0

set_variable ARG_URL1 ${JS_STR},4:25,"ppapi_geturl_success.html",7:0
set_variable ARG_URL1_FULL ${JS_STR},4:47,"http://localhost:5103/ppapi_geturl_success.html",1:0
set_variable ARG_URL2 ${JS_STR},4:32,"http://www.google.com/robots.txt"
set_variable ARG_URL3 ${JS_STR},4:39,"http://localhost:5103/non_existant.html",1:0

set_variable ARG_ERR1 ${JS_STR},4:38,"PPB_URLLoader::Open:\x20PP_ERROR_NOACCESS",2:0
set_variable ARG_ERR2 ${JS_STR},4:34,"PPB_URLRequestInfo::Create:\x20failed",6:0
set_variable ARG_GET  ${JS_STR},4:3,"GET",5:0

set_variable ARG_SUCC ${JS_STR},4:12,"hello\x20world!",4:0
set_variable ARG_0 ${JS_INT},4:0
set_variable ARG_1 ${JS_INT},4:1

set_variable WINDOW R(24,8:6,8:0x9f,8:0x5f29)
# TODO(robertm) figure out what this represents, maybe PP_Var
set_variable EXCEPTION R(8,8:0)
set_variable NULL R(8,8:0)

# Register  signatures found in all previous replay commands for export
# via install_upcalls
echo
echo "*** REGISTER REPLAY METHODS"
replay_activate

# This is not strictly necessary
echo
echo "*** INITIALIZE PEPPER"
initialize_pepper pepper_desc

# create a upcall service on the channel between nexe and sel_univeral
echo
echo "*** CREATE UPCALL SERVICE"
install_upcalls service_string

echo
echo "*** INIT MODULE"
replay 99 PPB_GetInterface:s:i  s("PPB_Core;0.3") * i(1)
replay 99 PPB_GetInterface:s:i  s("PPB_Instance;0.4") * i(1)
replay 99 PPB_GetInterface:s:i  s("PPB_Var(Deprecated);0.3") * i(1)
rpc PPP_InitializeModule i(0) ${MODULE} h(pepper_desc) s("${service_string}") * i(0) i(0)

echo
echo "*** INIT INSTANCE"
set_variable TAGS C(17,id\000name\000src\000type\000)
set_variable VALUES C(63,naclModule\000naclModule\000ppapi.nexe\000application/x-ppapi-nacl-srpc\000)
rpc PPP_Instance_DidCreate ${INSTANCE} i(4) ${TAGS} ${VALUES} * i(0)

echo
echo "*** TRIGGER REPAINT"
rpc PPP_Instance_DidChangeView ${INSTANCE} I(4,8,79,400,400) I(4,0,0,400,400) *
# leave time for flush of stdout

echo
echo "*** GET INSTANCE OBJECT"
rpc PPP_Instance_GetInstanceObject ${INSTANCE} * C(16)
# save result to variable
set_variable OBJECT ${result0}

echo
echo "*** HAS PROPERTY/METHOD (loadUrl)"
rpc HasProperty ${OBJECT} R(16,${ARG_loadUrl}) ${EXCEPTION} * i(0) C(8)
rpc HasMethod ${OBJECT} R(16,${ARG_loadUrl}) ${EXCEPTION} * i(0) C(8)

######################################################################
## URL1
######################################################################
echo
echo "*** CALL loadUrl(): url1 as buffer"
replay 1 PPB_GetInterface:s:i  s("PPB_URLRequestInfo;0.2") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLResponseInfo;0.1") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLLoader;0.1") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_FileIO(Dev);0.3") * i(1)

replay 1 PPB_FileIO_Dev_Create:i:i ${INSTANCE} * i(35)
replay 1 PPB_URLRequestInfo_Create:i:i ${INSTANCE} * i(30)
replay 1 PPB_URLLoader_Create:i:i  ${INSTANCE} * i(34)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(35) * i(1)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(35) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(35) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(35) * i(0)

replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(30) * i(1)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(30) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(30) * i(0)
replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(30) * i(0)

replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(34) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(34) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(34) * i(1)
replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(34) * i(0)

replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(0) R(40,${ARG_URL1}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(1) R(16,${ARG_GET}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(3) R(8,${ARG_0}) * i(1)

replay 1 PPB_URLLoader_Open:iii:i i(34) i(30) i(1) * i(-1)
rpc Call ${OBJECT} R(16,${ARG_loadUrl}) i(2) R(48,${ARG_URL1},${ARG_0}) ${NULL} * C(8) C(8)

echo
echo "*** async response for url1 part1"
replay 1 PPB_URLLoader_GetResponseInfo:i:i i(34) * i(38)
replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(38) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(38) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(38) * i(1)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(38) * i(0)
replay 1 PPB_URLResponseInfo_GetProperty:ii:C i(38) i(0) * R(56,${ARG_URL1_FULL})
replay 1 PPB_URLResponseInfo_GetProperty:ii:C i(38) i(3) * R(8,4:3,4:200)
replay 1 PPB_URLLoader_ReadResponseBody:iii:Ci i(34) i(1024) i(2) * C(0,) i(-1)
rpc RunCompletionCallback i(1) i(0) C(0,) *

echo
echo "*** async response for url1 part2"
replay 1 PPB_URLLoader_ReadResponseBody:iii:Ci i(34) i(1024) i(3) * C(0,) i(-1)
rpc RunCompletionCallback i(2) i(12) C(12,hello\x20world!) *

echo
echo "*** async response for url1 eof - report success"
replay 1 PPB_Instance_GetWindowObject:i:C ${INSTANCE} * ${WINDOW}
replay 1 Call:CCiCC:CC R(16,8:0x9f,8:0x5f29) R(24,${ARG_ReportResult}) i(4) R(96,${ARG_URL1_FULL},${ARG_0},${ARG_SUCC},${ARG_1}) ${NULL} * ${NULL} ${NULL}
replay 1 PPB_Core_ReleaseResource:i: i(30) *
replay 1 PPB_Core_ReleaseResource:i: i(34) *
replay 1 PPB_Core_ReleaseResource:i: i(35) *
replay 1 PPB_Core_ReleaseResource:i: i(38) *
rpc RunCompletionCallback i(3) i(0) C(0,) *

echo
echo "*** CALL loadUrl(): url1 as file"
replay 1 PPB_GetInterface:s:i  s("PPB_FileIO(Dev);0.3") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLRequestInfo;0.2") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLResponseInfo;0.1") * i(1)
replay 1 PPB_GetInterface:s:i  s("PPB_URLLoader;0.1") * i(1)

replay 1 PPB_URLLoader_Create:i:i  ${INSTANCE} * i(34)
replay 1 PPB_URLRequestInfo_Create:i:i ${INSTANCE} * i(30)
replay 1 PPB_FileIO_Dev_Create:i:i ${INSTANCE} * i(66)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(66) * i(1)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(66) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(66) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(66) * i(0)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(30) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(30) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(30) * i(1)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(30) * i(0)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(34) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(34) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(34) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(34) * i(1)

replay 1 PPB_Instance_GetWindowObject:i:C ${INSTANCE} * ${WINDOW}

replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(0) R(40,${ARG_URL1}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(1) R(16,${ARG_GET}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(30) i(3) R(8,${ARG_1}) * i(1)

replay 1 PPB_URLLoader_Open:iii:i i(34) i(30) i(4) * i(-1)

replay 1 PPB_Core_ReleaseResource:i: i(30) *
replay 1 PPB_Core_ReleaseResource:i: i(34) *
replay 1 PPB_Core_ReleaseResource:i: i(66) *

replay 1 Call:CCiCC:CC R(16,8:0x9f,8:0x5f29) R(24,${ARG_ReportResult}) i(4) R(80,${ARG_URL1},${ARG_1},${ARG_SUCC},${ARG_1}) ${NULL} * ${NULL} ${NULL}

rpc Call ${OBJECT} R(16,${ARG_loadUrl}) i(2) R(48,${ARG_URL1},${ARG_1}) ${NULL} * C(8) C(8)

######################################################################
## URL2
######################################################################
echo
echo "*** CALL loadUrl(): url2 as buffer - expected to fail"
replay 1 PPB_GetInterface:s:i  s("PPB_URLRequestInfo;0.2") * i(1)
replay 1 PPB_Instance_GetWindowObject:i:C ${INSTANCE} * ${WINDOW}
replay 1 Call:CCiCC:CC R(16,8:0x9f,8:0x5f29) R(24,${ARG_ReportResult}) i(4) R(104,${ARG_URL2},${ARG_0},${ARG_ERR2},${ARG_0}) ${NULL} * ${NULL} ${NULL}

replay 1 PPB_FileIO_Dev_Create:i:i ${INSTANCE} * i(35)
replay 1 PPB_URLRequestInfo_Create:i:i ${INSTANCE} * i(40)
replay 1 PPB_URLLoader_Create:i:i  ${INSTANCE} * i(44)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(35) * i(1)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(35) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(35) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(35) * i(0)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(40) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(40) * i(1)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(40) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(40) * i(0)

replay 1 PPB_FileIO_Dev_IsFileIO:i:i i(44) * i(0)
replay 1 PPB_URLRequestInfo_IsURLRequestInfo:i:i i(44) * i(0)
replay 1 PPB_URLResponseInfo_IsURLResponseInfo:i:i i(44) * i(0)
replay 1 PPB_URLLoader_IsURLLoader:i:i i(44) * i(1)

replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(40) i(0) R(40,${ARG_URL2}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(40) i(1) R(16,${ARG_GET}) * i(1)
replay 1 PPB_URLRequestInfo_SetProperty:iiC:i i(40) i(3) R(8,${ARG_0}) * i(1)

replay 1 PPB_URLLoader_Open:iii:i i(44) i(40) i(5) * i(-7)

replay 1 PPB_Instance_GetWindowObject:i:C ${INSTANCE} * ${WINDOW}
replay 1 Call:CCiCC:CC R(16,8:0x9f,8:0x5f29) R(24,${ARG_ReportResult}) i(4) R(104,${ARG_URL2},${ARG_0},${ARG_ERR1},${ARG_0}) ${NULL} * ${NULL} ${NULL}

replay 1 PPB_Core_ReleaseResource:i: i(35) *
replay 1 PPB_Core_ReleaseResource:i: i(40) *
replay 1 PPB_Core_ReleaseResource:i: i(44) *
rpc Call ${OBJECT} R(16,${ARG_loadUrl}) i(2) R(48,${ARG_URL2},${ARG_0}) ${NULL} * C(8) C(8)

######################################################################
## EPILOG
######################################################################
echo
echo "*** DONE"
rpc PPP_ShutdownModule *
