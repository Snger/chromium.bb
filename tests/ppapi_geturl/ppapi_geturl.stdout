
"*** REGISTER REPLAY METHODS"

"*** INITIALIZE PEPPER"

"*** CREATE UPCALL SERVICE"

"*** INIT MODULE"
--- Module::Module
replaying PPB_GetInterface:s:i:
output 0:  i(1)
replaying PPB_GetInterface:s:i:
output 0:  i(1)
replaying PPB_GetInterface:s:i:
output 0:  i(1)

"*** INIT INSTANCE"
--- PPP_GetInterface(PPP_Instance;0.4)
--- Module::GetPluginInterface(PPP_Instance;0.4)
--- Instance_DidCreate
PPP_Instance_DidCreate RESULTS:  i(1)

"*** TRIGGER REPAINT"
PPP_Instance_DidChangeView RESULTS:

"*** GET INSTANCE OBJECT"
--- Instance_GetInstanceObject

"*** HAS PROPERTY/METHOD (loadUrl)"
HasProperty RESULTS:  i(0)  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
HasMethod RESULTS:  i(1)  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)

"*** CALL loadUrl(): url1 as buffer"
--- ScriptableObject::LoadUrl('ppapi_geturl_success.html')
--- UrlLoadRequest::Load(as_file=0, 'ppapi_geturl_success.html')
replaying PPB_GetInterface:s:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_Create:i:i:
output 0:  i(30)
replaying PPB_GetInterface:s:i:
output 0:  i(1)
replaying PPB_GetInterface:s:i:
output 0:  i(1)
--- TestIsInterface: PPB_URLRequestInfo;0.2
replaying PPB_URLLoader_Create:i:i:
output 0:  i(34)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
output 0:  i(1)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
output 0:  i(0)
--- TestIsInterface: PPB_URLLoader;0.1
replaying PPB_URLLoader_IsURLLoader:i:i:
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
output 0:  i(1)
replaying PPB_URLLoader_Open:iii:i:
output 0:  i(-1)
Call RESULTS:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)

"*** async response for url1 part1"
--- UrlLoadRequest::OpenCallback
--- TestIsInterface: PPB_URLResponseInfo;0.1
replaying PPB_URLLoader_GetResponseInfo:i:i:
output 0:  i(38)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
output 0:  i(1)
replaying PPB_URLLoader_IsURLLoader:i:i:
output 0:  i(0)
replaying PPB_URLResponseInfo_GetProperty:ii:C:
output 0:  C(56,\x05\x00\x00\x00/\x00\x00\x00http://localhost:5103/ppapi_geturl_success.html\x00)
replaying PPB_URLResponseInfo_GetProperty:ii:C:
output 0:  C(8,\x03\x00\x00\x00\xc8\x00\x00\x00)
replaying PPB_URLLoader_ReadResponseBody:iii:Ci:
output 0:  C(0,)
output 1:  i(-1)
RunCompletionCallback RESULTS:

"*** async response for url1 part2"
--- UrlLoadRequest::ReadResponseBodyCallback
replaying PPB_URLLoader_ReadResponseBody:iii:Ci:
output 0:  C(0,)
output 1:  i(-1)
RunCompletionCallback RESULTS:

"*** async response for url1 eof - report success"
--- UrlLoadRequest::ReadResponseBodyCallback
--- ReportResult('http://localhost:5103/ppapi_geturl_success.html', as_file=0, 'hello world!', success=1)
replaying PPB_Instance_GetWindowObject:i:C:
output 0:  C(24,\x06\x00\x00\x00\x00\x00\x00\x00\x9f\x00\x00\x00\x00\x00\x00\x00\x29_\x00\x00\x00\x00\x00\x00)
replaying Call:CCiCC:CC:
output 0:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
output 1:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
replaying PPB_Core_ReleaseResource:i::
replaying PPB_Core_ReleaseResource:i::
replaying PPB_Core_ReleaseResource:i::
RunCompletionCallback RESULTS:

"*** CALL loadUrl(): url2 as buffer - expected to fail"
--- ScriptableObject::LoadUrl('http://www.google.com/robots.txt')
--- UrlLoadRequest::Load(as_file=0, 'http://www.google.com/robots.txt')
replaying PPB_URLRequestInfo_Create:i:i:
output 0:  i(40)
--- TestIsInterface: PPB_URLRequestInfo;0.2
replaying PPB_URLLoader_Create:i:i:
output 0:  i(44)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
output 0:  i(1)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
output 0:  i(0)
--- TestIsInterface: PPB_URLLoader;0.1
replaying PPB_URLLoader_IsURLLoader:i:i:
output 0:  i(0)
replaying PPB_URLRequestInfo_IsURLRequestInfo:i:i:
output 0:  i(0)
replaying PPB_URLResponseInfo_IsURLResponseInfo:i:i:
output 0:  i(0)
replaying PPB_URLLoader_IsURLLoader:i:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
output 0:  i(1)
replaying PPB_URLRequestInfo_SetProperty:iiC:i:
output 0:  i(1)
--- ReportResult('http://www.google.com/robots.txt', as_file=0, 'PPB_URLLoader::Open: PP_ERROR_NOACCESS', success=0)
replaying PPB_URLLoader_Open:iii:i:
output 0:  i(-7)
replaying PPB_Instance_GetWindowObject:i:C:
output 0:  C(24,\x06\x00\x00\x00\x00\x00\x00\x00\x9f\x00\x00\x00\x00\x00\x00\x00\x29_\x00\x00\x00\x00\x00\x00)
replaying Call:CCiCC:CC:
output 0:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
output 1:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)
replaying PPB_Core_ReleaseResource:i::
replaying PPB_Core_ReleaseResource:i::
Call RESULTS:  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)  C(8,\x00\x00\x00\x00\x00\x00\x00\x00)

"*** DONE"
--- PPP_ShutdownModule
PPP_ShutdownModule RESULTS:
